### ISSUE TITLE
Switch tranliteration/Roman system

### ISSUE THREAD

**Issue by PaulHeggarty** [lingdb/CoBL#380]
We need to switch from `Orthographic` and `Transliteration` fields to a more consistent equivalent, `Roman(ised)` vs. `Native Script`.  That is, the new `Roman(ised)` field will always contain texts in Roman characters, not any other script like Greek, Arabic, Devanāgarī and so on.  The new `Native Script` field will only be used for non-Roman scripts.
- [ ] That is, we need to run an 'update' query on the entire lexemes table.
* For all lexemes where the current `Transliteration` field is not blank (or not null), then the values in the `Transliteration` and  `Orthographic` fields need to be swapped with each other.  
* (Lexemes where the `Transliteration` field is currently blank just remain as they are.)
- [ ] Then rename all labels for these fields:  
* the original `Orthographic` needs to be renamed to `Roman(ised)`
* the original `Transliteration` needs to be renamed to `Native Script`
- [ ] Ideally, we should put a validation on texts entered in the  `Roman(ised)` field, to ensure that they are indeed Roman characters (although with accents, diacritics, and so on, and in many 'versions' of Roman script.  I’m not sure quite how valid this would be, and it's also not as high a priority point as the first two here.

**Comment by runjak** [comment]
I've made a list of characters currently entered into the `Roman(ised)` fields in the database:
```
" & ' ( ) * + , - . / 3 : ; = ? A B C D E F G H I J K L M N O P Q R S T U V W X Y Z [ ] ` a b c d e f g h i j k l m n o p q r s t u v w x y z ~ ¬ ° · Ä Ø ß à á â ã ä å æ ç è é ê ë ì í î ï ð ñ ò ó ô õ ö ø ù ú û ü ý þ ÿ ā ă ą ć ċ č ď đ ē ĕ ė ę ě ġ ģ ĩ Ī ī ĭ į ķ ļ ľ ł ń ņ ň ŋ ō œ ŕ ŗ ř ś ŝ ş Š š ţ ť Ũ ũ ū ŭ ů ų ŵ ŷ ź ż ž ſ ƕ ƫ ǔ ǖ ǘ ǝ ǣ ǧ ǫ ǰ ǵ ȓ ș ț ȫ ȳ ɐ ɑ ɒ ɔ ɖ ə ɛ ɜ ɡ ɣ ɨ ɪ ɫ ɭ ɳ ɵ ɽ ʁ ʃ ʈ ʉ ʊ ʌ ʒ ʓ ʰ ʷ ʸ ʹ ʺ ʼ ʾ ʿ ˈ ˊ ˌ ː ˑ ˜ ˤ ˳ ̀ ́ ̂ ̃ ̄ ̆ ̇ ̈ ̊ ̌ ̐ ̑ ̜ ̣ ̤ ̥ ̧ ̨ ̯ ̰ ̱ ̲ ̀ ́ ̈́ ͡ ή α β γ δ ε η θ ι κ λ μ ν ο π ρ ς σ τ υ φ χ ω ό ύ ώ а б в г д е ж з и й к л м н о п р с т у ф х ц ч ш щ ъ ы ь я ё є і ј ќ ў ѣ ѥ ѫ ѹ ҳ ӕ ա ե զ ը թ ի լ խ կ հ ղ ճ մ յ ն շ ո չ պ ջ ռ ս վ տ ր ց ւ ք آ ا ب ت ج د ر س ف ن و ُ ّ پ ک ھ आ ग छ त ध प ब भ र ल ा ि ी ु ू े ং এ ব ᴵ ᵛ ᵤ ḁ ḍ ḗ ḡ ḥ ḫ ḯ ḱ ḷ ṁ ṃ ṅ Ṇ ṇ ṓ ṕ ṙ ṛ ṡ ṣ ṭ ṯ ẏ ẓ ẖ ẛ ẽ ọ ỹ ἐ ὐ ή ῖ ‑ — ‘ ’ … ′ ″ ⁱ ⁽ ⁾ ₁ ₂ ₃ √ ≠
```
@PaulHeggarty I can use this list as a first check for validation, and make it so that we can easily adjust it later on.

**Comment by runjak** [comment]
@PaulHeggarty renaming the fields will also result in the `.csv` import expecting these columns:
```python
fieldConversions = {
    'id': intOrNone,
    'language_id': int,
    'meaning_id': int,
    'romanised': strOrEmpty,
    'phon_form': strOrEmpty,
    'gloss': strOrEmpty,
    'notes': strOrEmpty,
    '_order': intOrZero,
    'dubious': boolOrNone,
    'not_swadesh_term': boolOrNone,
    'phoneMic': strOrEmpty,
    'rfcWebLookup1': strOrEmpty,
    'rfcWebLookup2': strOrEmpty,
    'nativeScript': strOrEmpty,
    'lastEditedBy': strNonEmpty,
    'lastTouched': strToDatetime
}
```
Note that the columns are now named `romanised` and `nativeScript` instead of `transliteration` and `source_form`.

**Comment by runjak** [comment]
I've updated the way I'm checking the currently occuring `Romanised` fields in the database.
Results for only `Jena200` and `Current` are:
```
& ' ( ) * + , - . / : ; = ? A B C D E F G H I J K L M N O P Q R S T U V W X Y Z [ ] ` a b c d e f g h i j k l m n o p q r s t u v w x y z ~   ¬ · Ä Ø ß à á â ã ä å æ ç è é ê ë ì í î ï ð ñ ò ó ô õ ö ø ù ú û ü ý þ ÿ ā ă ą ć ċ č ď đ ē ĕ ė ę ě ġ ģ ĩ Ī ī ĭ į ķ ļ ľ ł ń ņ ň ŋ ō œ ŕ ŗ ř ś ŝ ş Š š ť ũ ū ŭ ů ų ŵ ŷ ź ż ž ſ ƕ ƫ ǔ ǖ ǘ ǝ ǣ ǧ ǫ ǰ ǵ ș ț ȳ ɑ ɔ ə ɛ ɜ ɡ ɣ ɨ ɪ ɫ ʁ ʈ ʉ ʊ ʌ ʒ ʰ ʷ ʸ ʹ ʺ ʼ ʾ ʿ ˈ ˊ ː ˑ ˜ ˤ ˳ ̀ ́ ̂ ̃ ̄ ̆ ̇ ̈ ̊ ̌ ̐ ̜ ̣ ̥ ̧ ̯ ̰ ̱ ̲ ̀ ́ ̈́ ͡ ή α β γ δ ε η θ ι κ λ μ ν ο ρ ς σ τ υ φ χ ω ό ύ ώ а д е и й к м н о п р с т х ш ъ ы ь я ӕ ա ե զ ը թ ի լ խ կ հ ղ մ յ ն շ ո պ ջ ռ ս վ տ ր ց ւ ք آ ا ب ت ج د ر س ف و ُ پ ھ ध प भ र ल ि ू ᴵ ᵛ ᵤ ḁ ḍ ḗ ḥ ḫ ḯ ḱ ḷ ṁ ṃ ṅ ṇ ṓ ṕ ṙ ṛ ṡ ṣ ṭ ṯ ẏ ẓ ẖ ẛ ẽ ọ ỹ ὐ ή ‑ — ‘ ’ ′ ″ ⁱ ₂ √ ≠
```
@PaulHeggarty I think we should compile a list of chars that we want to occur from this and compile a list of offending entries from it.

**Comment by cormacanderson** [comment]
There is something wrong with that list, @runjak , as there are Greek, Cyrillic, Arabic, and Devanagari characters included. Are you sure that that list doesn't refer to `nativeScript`?

**Comment by runjak** [comment]
@cormacanderson yes, I think these symbols actually occur in our current data in the romanised fields. I've compiled a list of these entries: [problematicLexemes.txt](https://github.com/lingdb/CoBL/files/666592/problematicLexemes.txt).
The list of symbols I've filtered for here can be seen in [here](https://github.com/lingdb/CoBL/commit/e0a5bc260251e0dff5f90430965e631f94d0bacb#diff-5bd5c2a7d79fbe857879773669321fa3R19).

**Comment by cormacanderson** [comment]
@runjak thanks for that. I've been through the list, changed some things, and asked language experts to check over others, where necessary. Could you please calculate the same list again, filtering for the same characters EXCEPT "ƕǝǣǧǫǰǵșțȳɑɔəɛɜɡɣɨɪɫʁʈʉʊʌʒ", "ъь", and "ḁẖ", which are either IPA or else special characters used in some still "Roman" orthographies. I'll then check over the revised ist.

**Comment by runjak** [comment]
I've redone the calculation with the requested changes.
The new file is here: 
[problematicLexemes.txt](https://github.com/lingdb/CoBL/files/668378/problematicLexemes.txt)

The number of entries is down from `1647` to `920`.

**Comment by cormacanderson** [comment]
The problematic characters appear to be <ſ> and <ẛ>, used in Slovenian (Early Modern), <ω> (also <ώ> (seemingly here with combining diactritic) in Rusyn, and also <‑>, acceptable as morpheme boundary, <ⁱ> for distinctive palatalisation. These can all be omitted, as well as <ó> I think. Perhaps you could try again with those omitted from the list.

**Comment by runjak** [comment]
I've redone the calculation.
The new file is here:
[problematicLexemes.txt](https://github.com/lingdb/CoBL/files/684509/problematicLexemes.txt)

The number of entries is down from `920` to `814`.
